web_extension/options/options_page.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+class OptionsSynchronizer {
+  constructor () {
+    this.options = new Options()
+  }
+
+  get checks () {
+    return document.querySelector('#photographic_options')
+                   .querySelectorAll('input[type=checkbox]')
+  }
+
+  get selects () {
+    return document.querySelector('#photographic_options')
+                   .querySelectorAll('select')
+  }
+
+  async read () {
+    let that = this
+    return this.options.read().then(function(){
```

Comment:
```suggestion
    const opts = await this.options.read()
    for (let check_box of this.checks) 
      [...]
```

Reply:
Oh hey look at that. I forgot to refactor this (or ...that) after I discovered I could just stash the promise and return it later.